app/services/imports/variable_definitions_service.rb

```diff
@@ -0,0 +1,58 @@
+require "csv"
+
+module Imports
+  class VariableDefinitionsService
+    attr_reader :path, :count
+
+    def initialize(path:)
+      @path = path
+      @count = 0
+    end
+
+    def call
+      files = Dir.glob(File.join(@path, "*.csv"))
+      files.each do |file|
+        process_file(file)
+      end
+    end
+
+  private
+
+    def process_file(file)
+      file_name = File.basename(file, ".csv")
+      parsed_file_name = file_name.split("_")
+      log_type = parsed_file_name[0]
+      user_type = parsed_file_name[1]
+      year = "20#{parsed_file_name[3]}".to_i
+
+      records_added = 0
+
+      CSV.foreach(file) do |row|
+        next if row.empty?
+
+        variable = row[0].downcase
+        definition = row[1..].join(",")
```

Comment:
Are there situations where definition is across multiple fields?

Reply:
There are situations where the definition is not wrapped in quotes " " and includes commas. The variable is taken as the all the content up to the first comma, the definition is built from the rest (if there are multiple parts).